//= output "DONE\n"

fun main() {
  do {
    try foo();
  } catch x: Array<Int> {
    assert(x.len() == 5);
    println("DONE");
  }
}

fun foo() throws {
  do {
    throw Array::<Int>(5, 1);
  } finally {
    forceCollect();
  }
}
